rero_ils/modules/stats/api.py

```diff
@@ -0,0 +1,277 @@
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Stats for pricing records."""
+import arrow
+from dateutil.relativedelta import relativedelta
+from invenio_search.api import RecordsSearch
+
+from ..acq_order_lines.api import AcqOrderLinesSearch
+from ..documents.api import DocumentsSearch
+from ..items.api import ItemsSearch
+from ..libraries.api import LibrariesSearch
+from ..loans.logs.api import LoanOperationLog
+from ..patrons.api import PatronsSearch
+
+
+class StatsForPricing:
+    """Statistics for pricing."""
+
+    def __init__(self):
+        """Constructor."""
+        self.now = arrow.utcnow()
+        self.months_delta = 3
+        _to = self.now.format(fmt='YYYY-MM-DDT23:59:59')
+        _from = (self.now - relativedelta(
+            months=self.months_delta)).format(fmt='YYYY-MM-DDT00:00:00')
+        self.date_range = {'gte': _from, 'lte': _to}
+
+    def get_all_libraries(self):
+        """Get all libraries in the system."""
+        return LibrariesSearch().source(['pid', 'name', 'organisation']).scan()
+
+    def stats(self):
+        """Compute all the statistics."""
+        for lib in self.get_all_libraries():
+            print(f'==== {lib.name} (pid: {lib.pid}) =====')
+            print('# docs', self.number_of_documents(lib.pid))
+            print('# libs', self.number_of_libraries(lib.organisation.pid))
+            print('# profs', self.number_of_librarians(lib.pid))
+            print('# active patrons', self.number_of_active_patrons(lib.pid))
+            print('# order lines', self.number_of_order_lines(lib.pid))
+            print('# checkouts', self.number_of_checkouts(lib.pid))
+            print('# renewals', self.number_of_renewals(lib.pid))
+            print(
+                '# statisfied ill requests',
+                self.number_of_satisfied_ill_request(lib.pid))
+
+            print(f'---- Optionals ----')
+            print('# items', self.number_of_items(lib.pid))
+            print('# new items', self.number_of_new_items(lib.pid))
+            print('# deleted items', self.number_of_deleted_items(lib.pid))
+            print('# patrons', self.number_of_patrons(lib.organisation.pid))
+            print(
+                '# new patrons',
+                self.number_of_new_patrons(lib.organisation.pid))
+            print('# checkins', self.number_of_checkins(lib.pid))
+            print('# requests', self.number_of_requests(lib.pid))
+            print('')
+
+    def number_of_documents(self, library_pid):
+        """Number of documents linked to my library.
+
+        point in time
```

Comment:
Is this `point in time` part of the docstring?

Reply:
Yes. This comes from the spec file.